src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,99 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_LENGTH = 10;
+    private T[] elementData;
+    private int size;
+
+    public ArrayList() {
+        elementData = (T[]) new Object[DEFAULT_LENGTH];
+        size = 0;
+    }
+
     @Override
     public void add(T value) {
-
+        checkForGrow();
+        elementData[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        checkIndex(index);
+        checkForGrow();
+        System.arraycopy(elementData, index, elementData, index + 1, size - index);
+        elementData[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index + 1);
+        return elementData[index];
     }
 
     @Override
     public void set(T value, int index) {
-
+        checkIndex(index + 1);
+        elementData[index] = value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index + 1);
+        final T object = elementData[index];
+        System.arraycopy(elementData, index + 1, elementData, index, size - index - 1);
+        size--;
+        elementData[size] = null;
+        return object;
     }
 
     @Override
     public T remove(T element) {
-        return null;
+        for (int i = 0; i < size; i++) {
+            if (elementData[i] == element
+                    || (elementData[i] != null) && elementData[i].equals(element)) {
+                return remove(i);
+            }
+        }
+        throw new NoSuchElementException("No Such Element");
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    public void grow() {
+        int newLength = (int) (elementData.length * 1.5);
```

Comment:
```suggestion
        int newLength = (int) (elementData.length * 1.5);
```
move magic number to constant